atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/persistentlock/PersistentLockService.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright 2017 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.persistentlock;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import com.palantir.atlasdb.keyvalue.api.CheckAndSetException;
+import com.palantir.atlasdb.keyvalue.api.KeyValueService;
+
+/**
+ * Provides endpoints for acquiring and releasing the Deletion Lock. This is intended to be used by backups and sweep,
+ * to ensure that only one of these operations is deleting data at once.
+ */
+@Path("/persistent-lock")
```

Comment:
Maybe I'm not understanding, but it's unclear to me then as a consumer of this endpoint how to tell the difference between successfully and failing to acquire a lock, because they both return a LockEntry.